src/rest/applications/mcm/resubmission.py

```diff
@@ -0,0 +1,428 @@
+        This is a customization for resetting a root request so
+        that it ends up in an 'approve/approved' status and could
+        be easily reused to recreate a chained request and inject it
+        again.
+
+        Args:
+            root_prepid: Root request's prepid
+
+        Returns:
+            An attribute describing how the root request was processed.
+        """
+        data: Union[dict, None] = self._mcm.get(
+            object_type="requests", object_id=root_prepid
+        )
+        if not data:
+            raise ValueError(f"Root request {root_prepid} not found")
+
+        approval = data.get("approval")
+        status = data.get("status")
+        if approval == "submit" and status == "done":
+            keep_output: Union[list[bool], None] = data.get("keep_output")
+            if not keep_output:
+                raise ValueError("Keep output is not defined")
+
+            if all(keep_output):
+                # Avoid resetting this kind of root request
+                # Just reuse it as it is
+                return RootRequestReset.KEEPS_OUTPUT
+            else:
+                self.logger.warning(
+                    (
+                        "Root request (%s) is already done, resetting it, "
+                        "announcing the invalidation, and setting it to approve/approved"
+                    ),
+                    root_prepid,
+                )
+                validation: Union[dict, None] = data.get("validation")
+                self.logger.debug("Validation results: %s", validation)
+                if not validation:
+                    raise ValueError("Validation results not available")
+
+                # Reset the request.
+                reset_result = self._mcm.reset(prepid=root_prepid)
+                if not reset_result:
+                    msg = f"Unable to reset the root request: {root_prepid}"
+                    self.logger.error(msg)
+                    raise RuntimeError(msg)
+
+                # Announce the invalidation.
+                self._invalidator._process_invalidation(request_prepids=[root_prepid])
+
+                # Force its status to approve/approved, re-including the validation.
+                data: dict = self._mcm.get(
+                    object_type="requests", object_id=root_prepid
+                )
+                assert isinstance(data, dict)
+
+                data["validation"] = validation
+                data["approval"] = "approve"
+                data["status"] = "approved"
+
+                updated_result = self._mcm.update(
+                    object_type="requests", object_data=data
+                )
+                if not updated_result or not updated_result.get("results"):
+                    msg = f"Error forcing the root request status: {root_prepid}"
+                    self.logger.error(msg)
+                    raise RuntimeError(msg)
+
+                return RootRequestReset.FULL_RESET
+        else:
```

Comment:
How does computing get notified of the changes in case the request is `submit/submit`? They need to stop processing and restart from scratch, does soft reset do this?

Reply:
Yes, it is handled by the soft reset operation (Details [here](https://github.com/cms-PdmV/cmsPdmV/blob/9d10daa2dc3d2801d7a6041147e24bc3b3fbc104/mcm/json_layer/request.py#L2642-L2654)).